### PR TITLE
SALTO-5474: fix idFields mechanism for newly created fields

### PR DESCRIPTION
**Pull request by edenhassid**
fix idFields mechanism for newly created fields

---

_Additional context for reviewer_

---
_Release Notes_: 
zendesk:
*  fix idFields mechanism for newly created fields

---
_User Notifications_: 
_Replace me with a short sentence that describes changes that will appear in NaCls and are not caused by user actions (e.g. a new annotation, field values that are converted to references, etc). Hidden changes should not be listed._

### CODE REVIEW

File: packages/zendesk-adapter/src/client/client.ts [hunk 1]
```diff
@@ -89,7 +163,9 @@ export default class ZendeskClient extends clientUtils.AdapterHTTPClient<
     args: clientUtils.ClientBaseParams,
   ): Promise<clientUtils.Response<clientUtils.ResponseValue | clientUtils.ResponseValue[]>> {
     try {
-      return await super.get(args)
+      // will be deleted once we move to new infra
```
Comment (adfineberg) [inline packages/zendesk-adapter/src/client/client.ts:166]: Do you have a ticket number for this?

**Review by adfineberg — APPROVED** [review]

File: packages/zendesk-adapter/src/client/client.ts (hunk 1 above)
Comment (edenhassid) [reply packages/zendesk-adapter/src/client/client.ts:166]: I'll check with @netama

File: packages/zendesk-adapter/src/client/client.ts [hunk 2]
```diff
@@ -35,6 +36,74 @@ type LogsFilterConfig = {
...
+  let usernameCounter = 0
+  let attachmentCounter = 0
+  return (
+    url: string,
+    response: clientUtils.Response<clientUtils.ResponseValue | clientUtils.ResponseValue[]>,
+  ): clientUtils.Response<clientUtils.ResponseValue | clientUtils.ResponseValue[]> => {
+    // for endpoint /api/v2/business_hours/schedules/{scheduleId}/holidays
+    if (url.includes('holidays') && isHolidayResponse(response)) {
+      response.data.holidays.forEach(day => {
+        day.start_year = holidayCounter
```
Comment (netama) [inline packages/zendesk-adapter/src/client/client.ts:86]: this one could just have the "final" values and we can remove the filter, right?

File: packages/zendesk-adapter/src/client/client.ts [hunk 3]
```diff
@@ -35,6 +36,74 @@ type LogsFilterConfig = {
...
+        day.end_year = holidayCounter
+        holidayCounter += 1
+      })
+      // for endpoint /api/v2/recipient_addresses
+    } else if (url.includes('recipient_addresses') && isSupportAddressResponse(response)) {
+      response.data.recipient_addresses.forEach(email => {
+        email.username = usernameCounter
```
Comment (netama) [inline packages/zendesk-adapter/src/client/client.ts:93]: same

File: packages/zendesk-adapter/src/client/client.ts (hunk 2 above)
Comment (edenhassid) [reply packages/zendesk-adapter/src/client/client.ts:86]: right, thought we could have this for now and change it later when the infra changes

**Pull request merged**